src/OMSimulatorLib/Types.h

```diff
@@ -70,12 +70,28 @@ typedef enum {
   oms_tlm_fine_grained
 } oms_tlm_interpolation_t;
 
+typedef enum {
+  oms_tlm_connector_type_state,
+  oms_tlm_connector_type_flow,
+  oms_tlm_connector_type_effort
+} oms_tlm_connector_type_enu_t;
+
 typedef enum {
   oms_solver_internal,         ///< internal solver; CS-FMU only
   oms_solver_explicit_euler,
   oms_solver_cvode
 } oms_solver_enu_t;
 
+typedef enum {
+  oms_element_none,
+  oms_element_model,      ///< composite model
+  oms_element_system,     ///< FMI or TLM system
+  oms_element_external, ///< External model
+  oms_element_fmu,      ///< FMU
+  oms_element_table,    ///< lookup table
+  oms_element_port      ///< port
+} oms3_element_type_enu_t;
```

Comment:
Since only systems and components are considered as elements, I propose the following enumeration:
```c
typedef enum {
  oms_element_none,
  oms_element_system_tlm,         ///< TLM System
  oms_element_system_wc,          ///< Weakly Coupled System
  oms_element_system_sc           ///< Strongly Coupled System
  oms_element_component_external, ///< External model
  oms_element_component_fmu,      ///< FMU
  oms_element_component_table     ///< lookup table
} oms3_element_type_enu_t;
```